Makefile

```
@@ -2,21 +2,25 @@
 # with Go source code. If you know what GOPATH is then you probably
 # don't need to bother with make.
 
-.PHONY: geth android ios evm all test clean libzkp
+.PHONY: geth android ios evm all test clean libzkp libzstd
 
 GOBIN = ./build/bin
 GO ?= latest
-GORUN = env GO111MODULE=on go run
+GORUN = go run
 
 libzkp:
 	cd $(PWD)/rollup/circuitcapacitychecker/libzkp && make libzkp
 
-nccc_geth: ## geth without circuit capacity checker
-	$(GORUN) build/ci.go install ./cmd/geth
+libzstd:
+	@wget https://github.com/scroll-tech/da-codec/releases/download/v0.0.0-rc0-ubuntu20.04/libzktrie.so
+	@wget https://github.com/scroll-tech/da-codec/releases/download/v0.0.0-rc0-ubuntu20.04/libscroll_zstd.so
+
+nccc_geth: libzstd ## geth without circuit capacity checker
+	@LD_LIBRARY_PATH="$(LD_LIBRARY_PATH):$(PWD)" CGO_LDFLAGS="-L$(PWD) -Wl,-rpath=$(PWD)" $(GORUN) build/ci.go install ./cmd/geth
```

Hmm does this work for you? I'm getting this:
```
/opt/homebrew/Cellar/go@1.21/1.21.10/libexec/pkg/tool/darwin_arm64/link: running cc failed: exit status 1
ld: unknown options: -rpath=/Users/peter/scroll/l2geth -rpath=/Users/peter/scroll/l2geth -rpath=/Users/peter/scroll/l2geth 
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

---

```suggestion
	@LD_LIBRARY_PATH="$(LD_LIBRARY_PATH):$(PWD)" CGO_LDFLAGS="-L$(PWD) -Wl,-rpath,$(PWD)" $(GORUN) build/ci.go install ./cmd/geth
```

Works like this, see https://stackoverflow.com/questions/26472590/ld-unknown-option-rpath-libs-linking-error-while-building-openframework

But I guess it's not possible to build it on arm, is it?

---

I think arm64 does not support `.so` files built from ubuntu20.04. Have not tried with `.dylib` built from an arm64 platform.

also alpine does not support `.so` files built from ubuntu20.04, too.

---

btw, do I need to accept the change request or it's just a discussion?